src/helpers/glucose-and-meals/steps.ts

```diff
@@ -0,0 +1,59 @@
+import MyDataHelps, { DeviceDataPointQuery, DeviceDataV2AggregateQuery } from '@careevolution/mydatahelps-js';
+import queryAllDeviceDataV2 from '../query-all-device-data-v2-aggregates';
+import { Reading } from './types';
+import { add, endOfDay, parseISO, startOfDay } from 'date-fns';
+import queryAllDeviceData from '../daily-data-providers/query-all-device-data';
+import { getMaxValueReadings } from './util';
+
+export function fitbitHalfHourStepsDataProvider(date: Date): Promise<Reading[]> {
+    const params: DeviceDataV2AggregateQuery = {
+        namespace: 'Fitbit',
+        type: 'activities-steps-intraday',
+        observedAfter: startOfDay(date).toISOString(),
+        observedBefore: endOfDay(date).toISOString(),
+        intervalAmount: 30,
+        intervalType: 'Minutes',
+        aggregateFunctions: ['sum']
+    };
+
+    return queryAllDeviceDataV2(params).then(aggregates => {
+        return aggregates.map(aggregate => {
+            return {
+                timestamp: add(parseISO(aggregate.date), { minutes: 15 }),
+                value: aggregate.statistics['sum']
+            };
+        });
+    }, () => []);
+}
+
+export function appleHealthHalfHourStepsDataProvider(date: Date): Promise<Reading[]> {
+    const params: DeviceDataPointQuery = {
+        namespace: 'AppleHealth',
+        type: 'HalfHourSteps',
+        observedAfter: startOfDay(date).toISOString(),
+        observedBefore: endOfDay(date).toISOString(),
+    };
+
+    return queryAllDeviceData(params).then(dataPoints => {
+        return dataPoints.map(dataPoint => {
+            return {
+                timestamp: add(parseISO(dataPoint.observationDate!), { minutes: -15 }),
+                value: parseInt(dataPoint.value)
+            };
+        });
+    }, () => []);
+}
+
+export async function getSteps(date: Date): Promise<Reading[]> {
+    let providers: Promise<Reading[]>[] = [];
+
+    let settings = await MyDataHelps.getDataCollectionSettings();
+    if (settings.fitbitEnabled) {
+        providers.push(fitbitHalfHourStepsDataProvider(date));
+    }
+    if (settings.queryableDeviceDataTypes.find(s => s.namespace == 'AppleHealth' && s.type == 'HalfHourSteps')) {
+        providers.push(appleHealthHalfHourStepsDataProvider(date));
+    }
+
+    return providers.length > 0 ? await getMaxValueReadings(providers) : [];
```

Comment:
Similar question here as above regarding not looking at Garmin data?

Reply:
I think we do want to support Garmin for this too, I just need to work out how best to do that.  For Fitbit, we are using the V2 API and aggregation.  For Apple Health, we use the special statistics query rule configuration.  I think support Garmin data was just recently added to the V2 real time API, so I just need to look at that a bit and add a "half hour steps" computation.

Reply:
I've added the Garmin half hour steps implementation.